deployments/ocis-monitoring/grafana-agent/templates/grafana-agent.yaml

```diff
@@ -0,0 +1,171 @@
+---
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: grafana-agent
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: grafana-agent
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  - nodes/proxy
+  - nodes/metrics
+  - services
+  - endpoints
+  - pods
+  - events
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - networking.k8s.io
+  resources:
+  - ingresses
+  verbs:
+  - get
+  - list
+  - watch
+- nonResourceURLs:
+  - /metrics
+  - /metrics/cadvisor
+  verbs:
+  - get
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: grafana-agent
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: grafana-agent
+subjects:
+- kind: ServiceAccount
+  name: grafana-agent
+  namespace: grafana-agent
+
+---
+apiVersion: monitoring.grafana.com/v1alpha1
+kind: GrafanaAgent
+metadata:
+  name: grafana-agent-metrics
+  labels:
+    app: grafana-agent-metrics
+spec:
+  {{- with .Values.imagePullSecrets }}
+  imagePullSecrets:
+    {{- toYaml . | nindent 8 }}
+  {{- end }}
+  # resources:
+  #   requests:
+  #     cpu: 500m
+  #     memory: 11Gi
+  image: "{{ .Values.image.repository }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
+  logLevel: info
+  serviceAccountName: grafana-agent
+  metrics:
+    instanceSelector:
+      matchLabels:
+        agent: grafana-agent-metrics
+    externalLabels:
+      cluster: {{ .Values.clusterName }}
+
+  integrations:
+    selector:
+      matchLabels:
+        agent: grafana-agent-integrations
+
+---
+apiVersion: monitoring.grafana.com/v1alpha1
+kind: GrafanaAgent
+metadata:
+  name: grafana-agent-logs
+  labels:
+    app: grafana-agent-logs
+spec:
+  {{- with .Values.imagePullSecrets }}
+  imagePullSecrets:
+    {{- toYaml . | nindent 8 }}
+  {{- end }}
+  image: "{{ .Values.image.repository }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
+  logLevel: info
+  resources:
+    requests:
+      cpu: 100m
+      memory: 300Mi
+  serviceAccountName: grafana-agent
+
+  logs:
+    instanceSelector:
+      matchLabels:
+        agent: grafana-agent-logs
+    clients:
+      # default value, is overwritten in LogsInstance
+      - url: http://{{ .Values.remoteLokiHost }}/loki/api/v1/push
```

Comment:
I'm trying it in minkube and having trouble on the `grafana-agent-logs-logs` pod:

```
grafana-agent-logs-logs-wjhr4 grafana-agent ts=2024-02-16T15:04:40.879266992Z caller=client.go:430 level=error component=logs logs_config=grafana-agent/primary component=client host=loki.kube.owncloud.test msg="final error sending batch" status=308 tenant= error="server returned HTTP status 308 Permanent Redirect (308): <html>"
```

Probably this might be a HTTP -> HTTPS redirect?

Curl at least supports that:

```
curl http://loki.kube.owncloud.test -vv 
* Host loki.kube.owncloud.test:80 was resolved.
* IPv6: (none)
* IPv4: 192.168.49.2
*   Trying 192.168.49.2:80...
* Connected to loki.kube.owncloud.test (192.168.49.2) port 80
> GET / HTTP/1.1
> Host: loki.kube.owncloud.test
> User-Agent: curl/8.5.0
> Accept: */*
> 
< HTTP/1.1 308 Permanent Redirect
< Date: Fri, 16 Feb 2024 15:17:15 GMT
< Content-Type: text/html
< Content-Length: 164
< Connection: keep-alive
< Location: https://loki.kube.owncloud.test
< 
<html>
<head><title>308 Permanent Redirect</title></head>
<body>
<center><h1>308 Permanent Redirect</h1></center>
<hr><center>nginx</center>
</body>
</html>
* Connection #0 to host loki.kube.owncloud.test left intact
*
```

Turns out I need to explicitly turn off the HTTP -> HTTPS redirect by adding `ssl-redirect: "false"` on `kubectl -n ingress-nginx edit cm/ingress-nginx-controller`